services/core/SEP2Agent/sep2/agent.py

```diff
@@ -0,0 +1,570 @@
+    """
+    try:
+        config = utils.load_config(config_path)
+    except StandardError:
+        config = {}
+
+    if not config:
+        _log.info("Using SEP2 Agent defaults for starting configuration.")
+
+    agentid = config.get('agentid', 'SEP2-agent')
+    devices = config.get('devices', [])  # To add devices, include them in a config file
+    sep2_server_sfdi = config.get('sep2_server_sfdi', 'foo')  # This default should be overridden in config file
+    sep2_server_lfdi = config.get('sep2_server_lfdi', 'bar')  # This defauly should be overridden in config file
+    load_shed_device_category = config.get('load_shed_device_category', '0020')
+    timezone = config.get('timezone', 'America/Los_Angeles')
+
+    return SEP2Agent(agentid,
+                     devices,
+                     sep2_server_sfdi,
+                     sep2_server_lfdi,
+                     load_shed_device_category,
+                     timezone,
+                         **kwargs)
+
+
+class SEP2Agent(Agent):
+    """
+        Agent that handles SEP2 communication.
+
+        SEP2Agent uses the VOLTTRON web service to communicate with SEP2 end devices.
+        End device configuration is outlined in the agent config file.
+
+        SEP2 data is exposed via get_point(), get_points() and set_point() calls.
+        A SEP2 device driver (sep2.py under MasterDriverAgent) can be configured,
+        which gets and sets data by sending RPCs to this agent.
+
+        For further information about this subsystem, please see the VOLTTRON
+        SEP 2.0 DER Support specification, which is located in VOLTTRON readthedocs
+        under specifications/sep2_agent.html.
+
+        This agent can be installed as follows:
+            export VIP_SOCKET="ipc://$VOLTTRON_HOME/run/vip.socket"
+            export SEP2_ROOT=$VOLTTRON_ROOT/services/core/SEP2Agent
+            cd $VOLTTRON_ROOT
+            python scripts/install-agent.py -s $SEP2_ROOT -i sep2agent -c $SEP2_ROOT/sep2.config -t sep2agent -f
+    """
+
+    def __init__(self, agentid='SEP2-agent', device_config=[], sep2_server_sfdi='foo', sep2_server_lfdi='bar',
+                 load_shed_device_category='0020', timezone='America/Los_Angeles', **kwargs):
+        super(SEP2Agent, self).__init__(enable_web=True, **kwargs)
+
+        self.agentid = agentid
+        self.device_config = device_config
+        self.sep2_server_sfdi = sep2_server_sfdi
+        self.sep2_server_lfdi = sep2_server_lfdi
+        self.load_shed_device_category = load_shed_device_category
+        self.timezone = timezone
+
+        self.default_config = {"agentid": agentid,
+                               "device_config": device_config,
+                               "sep2_server_sfdi": sep2_server_sfdi,
+                               "sep2_server_lfdi": sep2_server_lfdi,
+                               "load_shed_device_category": load_shed_device_category,
+                               "timezone": timezone}
+
```

Comment:
Thanks for the recommendations, Kyle, they are much appreciated. I've implemented all of the other changes you suggested here, and I've added self.vip.config.set_default("config", self.default_config), but I wasn't able to eliminate the redundant calls to register_endpoints() and register_devices(). If I do that, the pytests fail.